pkg/cmd/gist/view/view.go

```diff
@@ -74,7 +74,7 @@ func viewRun(opts *ViewOptions) error {
 	u, err := url.Parse(opts.Selector)
 	if err == nil {
 		if strings.HasPrefix(u.Path, "/") {
-			gistID = u.Path[1:]
+			gistID = strings.Split(u.Path, "/")[2]
```

Comment:
Before trying to select an element in the slice with index `2`, it's good to first check whether the slice is large enough to begin with, and return a descriptive error otherwise. This will prevent a runtime crash, which is rarely a good experience for users, since crash messages are hard to decipher.

Reply:
What would be the right way to send a descriptive error like `Invalid gist URL https://....` to stdout?
This works and allows us to check the error in tests but I'm not sure if it's ok to return nil.

```go
if strings.Contains(gistID, "/") {
	u, err := url.Parse(opts.Selector)
	if err == nil && strings.HasPrefix(u.Path, "/") {
		split := strings.Split(u.Path, "/")
                                                                  
		if len(split) > 2 && split[2] != "" {
			gistID = strings.Split(u.Path, "/")[2]
		} else if opts.IO.IsStderrTTY() {
			fmt.Fprintf(opts.IO.ErrOut, "Invalid gist URL %s", u)
			return nil
		}
	}
}

```